### ISSUE TITLE
Signed-right-shift with large shift amounts gives wrong result

### ISSUE THREAD

**Issue by brianhuffman** [GaloisInc/cryptol#716]
The signed right shift operator `(>>$)` produces the wrong result when given a shift amount too large to fit in a Haskell `Int` (2^63 or higher).
```
Cryptol> 0x55 >>$ 0x8000000000000000
0x55
```
Both the concrete and symbolic evaluation backends have this problem:
```
Cryptol> :prove \(x:[8]) -> x >>$ 0x8000000000000000 == x
Q.E.D.
(Total Elapsed Time: 0.008s, using "Z3")
Cryptol> :exhaust \(x:[8]) -> x >>$ 0x8000000000000000 == x
Using exhaustive testing.
Passed 256 tests.
Q.E.D.
```
Here is the offending function from the concrete backend. It includes an unsafe use of `fromInteger` (#637).
https://github.com/GaloisInc/cryptol/blob/7e841e915f34a21bbfad67070c7338662ac28601/src/Cryptol/Eval/Concrete.hs#L279-L288

**Comment by brianhuffman** [comment]
The situation is a bit complicated in the symbolic backend. It appears that the bug only shows up when the shift amount is concrete.
```
Cryptol> :prove \(i:[64]) -> (0x55 >>$ i == 0x55) == (i == 0)
Q.E.D.
(Total Elapsed Time: 0.010s, using "Z3")
```

**Comment by robdockins** [comment]
Fixed via #724.
